Sources/StreamChat/Controllers/PollController/PollController.swift

```diff
@@ -0,0 +1,277 @@
+            log.warning("Callback called while self is nil")
+            return nil
+        }
+        
+        let observer = environment.pollObserverBuilder(
+            StreamRuntimeCheck._isBackgroundMappingEnabled,
+            self.client.databaseContainer,
+            PollDTO.fetchRequest(for: pollId),
+            { try $0.asModel() as Poll },
+            NSFetchedResultsController<PollDTO>.self
+        )
+        .onChange { [weak self] change in
+            self?.delegateCallback { [weak self] delegate in
+                guard let self = self else {
+                    log.warning("Callback called while self is nil")
+                    return
+                }
+                delegate.pollController(self, didUpdatePoll: change)
+            }
+        }
+
+        return observer
+    }()
+    
+    private(set) lazy var ownVotesObserver: ListDatabaseObserverWrapper<PollVote, PollVoteDTO> = {
+        let request = PollVoteDTO.pollVoteListFetchRequest(query: self.ownVotesQuery)
+
+        let observer = environment.ownVotesObserverBuilder(
+            StreamRuntimeCheck._isBackgroundMappingEnabled,
+            self.client.databaseContainer,
+            request,
+            { try $0.asModel() }
+        )
+        
+        observer.onDidChange = { [weak self] changes in
+            self?.delegateCallback { [weak self] delegate in
+                guard let self = self else {
+                    log.warning("Callback called while self is nil")
+                    return
+                }
+
+                delegate.pollController(self, didUpdateCurrentUserVotes: changes)
+            }
+        }
+
+        return observer
+    }()
+    
+    var _basePublishers: Any?
+    /// An internal backing object for all publicly available Combine publishers. We use it to simplify the way we expose
+    /// publishers. Instead of creating custom `Publisher` types, we use `CurrentValueSubject` and `PassthroughSubject` internally,
+    /// and expose the published values by mapping them to a read-only `AnyPublisher` type.
+    @available(iOS 13, *)
+    var basePublishers: BasePublishers {
+        if let value = _basePublishers as? BasePublishers {
+            return value
+        }
+        _basePublishers = BasePublishers(controller: self)
+        return _basePublishers as? BasePublishers ?? .init(controller: self)
+    }
+    
+    private let environment: Environment
+        
+    init(client: ChatClient, messageId: MessageId, pollId: String, environment: Environment = .init()) {
+        self.client = client
+        self.messageId = messageId
+        self.pollId = pollId
+        self.environment = environment
+        ownVotesQuery = PollVoteListQuery(
+            pollId: pollId,
+            optionId: nil,
+            pagination: .init(pageSize: 25, cursor: nil),
+            filter: .and(
+                [.equal(.userId, to: client.currentUserId ?? ""), .equal(.pollId, to: pollId)]
+            )
+        )
```

Comment:
This query fetches the current user's votes. The backend allows max of 10 votes per user, but I put it to the max of the query just in case. Having this configurable shouldn't be possible for customers.